include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -8874,6 +8874,18 @@ def err_bounds_type_annotation_lost_checking : Error<
   def err_expected_bounds : Error<
     "expression has no bounds">;
 
+  def err_expected_bounds_for_ptr_cast : Error<
+    "expression has no bounds"
+    ", cast to ptr<T> expects source to have bounds">;
+
+  def err_expected_bounds_for_assignment : Error<
+    "expression has no bounds"
+    ", right-hand side of assignment expected to have bounds because the left-hand side has bounds">;
+
+  def err_expected_bounds_for_initializer : Error<
+    "expression has no bounds"
+    ", an initializer expected to have bounds because variable declaration has bounds">;
```

Comment:
Actually, this modification makes changes in expected-error message in make check-clang
So, I just have added the reason why this expected bounds error occurrs at the end of "expression has no bounds"
After modifying this error message, I will also modify related expected error message